allennlp/commands/dry_run.py

```diff
@@ -0,0 +1,199 @@
+        logger.info(f"You passed a directory_path in your config which already exists and is non-empty. "
+                    f"Refusing to overwrite - we saved it to {serialization_dir} instead.")
+    elif vocab_dir is not None:
+        logger.info(f"You passed a directory_path in your config which was empty. Also "
+                    f"writing the vocabulary to {vocab_dir}.")
+        vocabulary.save_to_files(vocab_dir)
+
+
+def verbosely_create_vocabulary(vocab_params: Params, instances: List[Instance]) -> Vocabulary:
+    """
+    Given a parameter config specifying a vocabulary and a list of instances, this function
+    creates a Vocabulary from the instances. Additionally, it logs corpus statistics, prints
+    a random selection of instances and prints vocabulary sizes and namespaces.
+
+    Parameters
+    ----------
+    vocab_params : ``Params``, required
+        The parameters for the Vocabulary we create.
+    instances : ``List[Instance]``, required.
+        The instances to build the vocabulary from.
+
+    Returns
+    -------
+    The created Vocabulary.
+    """
+    min_count = vocab_params.pop("min_count", None)
+    max_vocab_size = vocab_params.pop_int("max_vocab_size", None)
+    non_padded_namespaces = vocab_params.pop("non_padded_namespaces", DEFAULT_NON_PADDED_NAMESPACES)
+    pretrained_files = vocab_params.pop("pretrained_files", {})
+    only_include_pretrained_words = vocab_params.pop_bool("only_include_pretrained_words", False)
+    vocab_params.assert_empty("Vocabulary - from dataset")
+
+    logger.info("Fitting token dictionary from dataset.")
+    namespace_token_counts: Dict[str, Dict[str, int]] = defaultdict(lambda: defaultdict(int))
+    sequence_field_lengths: Dict[str, List] = defaultdict(list)
+
+    for instance in instances:
+        instance.count_vocab_items(namespace_token_counts)
+    vocabulary = Vocabulary(counter=namespace_token_counts,
+                            min_count=min_count,
+                            max_vocab_size=max_vocab_size,
+                            non_padded_namespaces=non_padded_namespaces,
+                            pretrained_files=pretrained_files,
+                            only_include_pretrained_words=only_include_pretrained_words)
+
+    for instance in instances:
+        instance.index_fields(vocabulary)
+        for field, field_padding_lengths in instance.get_padding_lengths().items():
+            for key, value in field_padding_lengths.items():
+                sequence_field_lengths[f"{field}.{key}"].append(value)
+
+    print("\n\n----Dataset Statistics----\n")
+    for name, lengths in sequence_field_lengths.items():
+        print(f"Statistics for {name}:")
+        print(f"\tLengths: Mean: {numpy.mean(lengths)}, Standard Dev: {numpy.std(lengths)}, "
+              f"Max: {numpy.max(lengths)}, Min: {numpy.min(lengths)}")
+
+    print("\n10 Random instances: ")
+    for i in list(numpy.random.randint(len(instances), size=10)):
```

Comment:
Really doesn't matter (i.e., feel free to totally ignore this), but maybe slightly more pythonic:

```python
for i, instance in enumerate(random.sample(instances, 10)):
    ...
```

Oh, except you're outputting the original index, not "first random example", "second random example", etc. ...  Ok, yeah, what I said won't do that.